### PR TITLE
fix remove weekends

### PR DESCRIPTION
**Pull request by kamigerami**
fixes https://github.com/codelabsab/timereport-slack/issues/98

### CODE REVIEW

**Comment by kamigerami** [comment]
Please verify @parberge @tommymalmqvist  before merge to master.

**Comment by tommymalmqvist** [comment]
Kanske vi ska städa bort console.log?

File: src/components/TableBody.js [hunk 1]
```diff
@@ -11,6 +14,21 @@ class TableBody extends Component {
     var lockstate = this.props.lockstate || 'unlocked';
     var total_working_hours = this.props.totaldays * 8;
     var total = 0;
+    // clear weekends from data
+    var weekday = [];
+    var i = 0;
+    Object.values(data).forEach(value => {
+      console.log('index is : ' + i)
```
Comment (tommymalmqvist) [inline src/components/TableBody.js]: Städa bort den här?

File: src/components/TableBody.js [hunk 2]
```diff
@@ -11,6 +14,21 @@ class TableBody extends Component {
...
+      if (moment(value.event_date).isoWeekday() < 6) {
+      } else {
+        weekday.push(i)
+      }
+      i++;
+    });
+    console.log('weekday array is ' + weekday)
```
Comment (tommymalmqvist) [inline src/components/TableBody.js]: This one too

Comment (kamigerami) [reply src/components/TableBody.js]: fixed in https://github.com/codelabsab/timereport-ui/pull/8/commits/968d0e637092d5286ac590562b5526da9fe0d1c9

File: src/components/TableBody.js (hunk 1 above)
Comment (kamigerami) [reply src/components/TableBody.js]: fixed in https://github.com/codelabsab/timereport-ui/pull/8/commits/968d0e637092d5286ac590562b5526da9fe0d1c9

File: src/components/TableBody.js [hunk 3]
```diff
@@ -11,12 +14,21 @@ class TableBody extends Component {
     var lockstate = this.props.lockstate || 'unlocked';
     var total_working_hours = this.props.totaldays * 8;
     var total = 0;
+    // clear weekends from data
+    var weekday = [];
+    var i = 0;
+    Object.values(data).forEach(value => {
+      if (moment(value.event_date).isoWeekday() < 6) {
```
Comment (parberge) [inline src/components/TableBody.js]: Maybe being picky, but why not just `moment(value.event_date).isoWeekday() > 6` and push when that occurs? (I guess push is to continue the loop with next item?)

Comment (kamigerami) [reply src/components/TableBody.js]: you mean to skip the Object.values(data).forEach(value => { ?

Comment (kamigerami) [reply src/components/TableBody.js]: oh nevermind! got what you meant :) yes that is absolutely right

File: src/components/TableBody.js [hunk 4]
```diff
@@ -11,12 +14,21 @@ class TableBody extends Component {
...
+      } else {
+        weekday.push(i)
+      }
+      i++;
+    });
+    for (var idx = weekday.length -1; idx >= 0; idx--)
```
Comment (parberge) [inline src/components/TableBody.js]: Just for curiosity: What does `idx` mean?

It looks like a loop that goes from a number down to 0?

File: src/components/TableBody.js [hunk 5]
```diff
@@ -11,12 +14,21 @@ class TableBody extends Component {
...
+    data.splice(weekday[idx],1);
+
```
Comment (parberge) [inline src/components/TableBody.js]: Sorry for my lack of understanding, but what does this do?

File: src/components/TableBody.js (hunk 4 above)
Comment (kamigerami) [reply src/components/TableBody.js]: yes just index.. my own variable

File: src/components/TableBody.js (hunk 5 above)
Comment (kamigerami) [reply src/components/TableBody.js]: splice will remove those matching indices from the object.

Comment (kamigerami) [reply src/components/TableBody.js]: so we are looping through the "indexes" that we matched previously (for the dates that are weekends) and pushed to wekend array through weekend.push. then we are using those indexes to identify what to remove from the data Object.

File: src/components/TableBody.js (hunk 3 above)
Comment (kamigerami) [reply src/components/TableBody.js]: fixed in : https://github.com/codelabsab/timereport-ui/pull/8/commits/442a9e15e23f50102792ff61305a1e3b512ca124

thnx

File: src/components/TableBody.js (hunk 5 above)
Comment (parberge) [reply src/components/TableBody.js]: Not sure I understand, but OK 😼

**Comment by parberge** [comment]
Does this look correct?
image

And:
image

To me the summary looks wierd

**Comment by kamigerami** [comment]
seems fine now?
Screen Shot 2019-09-03 at 09 24 48
Screen Shot 2019-09-03 at 09 25 01

**Comment by kamigerami** [comment]
nothing seems wrong with february. -152 reported hours out of 160 leaves 8 total for the month.

**Pull request merged**